tools/eval/datasets/otb2015.py

```diff
@@ -0,0 +1,333 @@
+            for attr_ in v.attr:
+                self.attr[attr_].append(k)
+
+    def __getitem__(self, idx):
+        if isinstance(idx, str):
+            return self.videos[idx]
+        elif isinstance(idx, int):
+            sorted_keys = sorted(list(self.videos.keys()))
+            return self.videos[sorted_keys[idx]]
+
+    def __len__(self):
+        return len(self.videos)
+
+    def __iter__(self):
+        sorted_keys = sorted(list(self.videos.keys()))
+        for key in sorted_keys:
+            yield self.videos[key]
+
+def get_axis_aligned_bbox(region):
+    """Converts a region to (cx, cy, w, h) representing an axis-aligned box."""
+    nv = region.size
+    if nv == 8:
+        cx = np.mean(region[0::2])
+        cy = np.mean(region[1::2])
+        x1 = min(region[0::2])
+        x2 = max(region[0::2])
+        y1 = min(region[1::2])
+        y2 = max(region[1::2])
+        A1 = np.linalg.norm(region[0:2] - region[2:4]) * np.linalg.norm(region[2:4] - region[4:6])
+        A2 = (x2 - x1) * (y2 - y1)
+        s = np.sqrt(A1 / A2)
+        w = s * (x2 - x1) + 1
+        h = s * (y2 - y1) + 1
+    else:
+        x, y, w, h = region
+        cx = x + w / 2
+        cy = y + h / 2
+    return cx, cy, w, h
+
+class OTB2015:
+    def __init__(self, root):
+        # Go up one if directory is provided
+        root = os.path.abspath(root)
+        if root.endswith("OTB2015"):
+            root = os.path.dirname(root)
+        print(root)
+
+        # Unzip the OTB100.zip file
+        if os.path.exists(f'{root}/OTB100.zip'):
+            os.system(f'unzip -q "{os.path.join(root, "OTB100.zip")}" -d "{root}"')
+            os.remove(f'{root}/OTB100.zip')
+
+        # Move the JSON label in if it's outside
+        if os.path.exists(f'{root}/OTB.json'):
+            os.rename(f'{root}/OTB.json', f'{root}/OTB100/OTB.json')
+
+        if os.path.exists(f'{root}/OTB100'):
+            original_directories = ['Jogging', 'Skating2', 'Human4']
+            updated_directories = ['Jogging-1', 'Jogging-2', 'Skating2-1', 'Skating2-2', 'Human4-2', 'OTB.json']
+            original_exist = all(os.path.exists(f'{root}/OTB100/{dir}') for dir in original_directories)
+            updated_exist = all(os.path.exists(f'{root}/OTB100/{dir}') for dir in updated_directories)
+            if original_exist:
+                os.rename(f'{root}/OTB100/Jogging', f'{root}/OTB100/Jogging-1')
+                os.rename(f'{root}/OTB100/Skating2', f'{root}/OTB100/Skating2-1')
+                os.rename(f'{root}/OTB100/Human4', f'{root}/OTB100/Human4-2')
+                os.system(f'cp -r "{root}/OTB100/Jogging-1" "{root}/OTB100/Jogging-2"')
+                os.system(f'cp -r "{root}/OTB100/Skating2-1" "{root}/OTB100/Skating2-2"')
+            elif not updated_exist:
+                raise RuntimeError("Not all files needed for setup are present.")
+
+        self.root = f'{root}/OTB2015'
+        self.dataset = OTBDATASET(self.root)
```

Comment:
Please drop any renaming or moving operations.

- Since the data package uses `OTB100` as filename, `otb100` should be used in the code and documentation to keep it aligned.
- Do not move json. You should add description in README and specify the directory architecture with `tree`. Lets say user should create directory `otb`, then `OTB.json` and `OTB100.zip` should be put in `otb`. User should extract `OTB100.zip` manually.
- Why does the original data need to be renamed and copied?